### ISSUE TITLE
Cloudstate Build – building node modules

### ISSUE THREAD

**Issue by marcellanz** [cloudstateio/cloudstate#323]
_(I have titled this issue first "it:test fails with node 14.x on macOS", but its more than that)_

**tl;dr**

- node-support does not build with node version >12.x (on macOS)
- `sbt it:test` fails if `cd samples/js-shopping-cart && npm install && cd -` has not been run.
- both is not documented in the [Cloudstate Build](https://cloudstate.io/docs/developer/thebuild.html)

I'm not a "node guy" so what I describe is a non-issue perhaps :)

---
The [Cloudstate Build](https://cloudstate.io/docs/developer/thebuild.html) mentions to issue `npm install` in node-support to "build the documentation using the paradox". 

Having a new laptop I recently tried to build a fresh checked out cloudstate repository and built it like described in the [Cloudstate Build](https://cloudstate.io/docs/developer/thebuild.html):

```
> cd node-support && npm install && cd -
```
fails fast with

```
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp WARN Tried to download(404): https://node-precompiled-binaries.grpc.io/grpc/v1.24.2/node-v83-darwin-x64-unknown.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for grpc@1.24.2 and node@14.2.0 (node-v83 ABI, unknown) (falling back to source compile with node-gyp) 
....
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! grpc@1.24.2 install: `node-pre-gyp install --fallback-to-build --library=static_library`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the grpc@1.24.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

This error is reported  by:
- https://github.com/nodejs/node-gyp/blob/master/macOS_Catalina.md
- [https://github.com/grpc/grpc-node/issues/1376](https://github.com/grpc/grpc-node/issues/1376)

I have installed node using `brew install node` and it delivers me v14 of node which obviously does not work with the grpc version we use. I think I once built it from source by accident and followed the recommendation from the first link above by re-installing XCode tools; but this time I did not.

So having a fresh Mac and java, sbt and node installed I get this trying to build cloudstate.

**Issue**
Once I've installed v12.x of node, I could build the `node-support` module. And once I built `samples/js-shopping-cart` then `sbt it:test` ran successfully.
`cloudstate/.travis.yml` shows clearly what to do:
https://github.com/cloudstateio/cloudstate/blob/master/.travis.yml#L54-L55
to run the integration tests and therefore the TCK.

**Suggested Resolution**
- I think we can document the node version needed for the [Cloudstate Build](https://cloudstate.io/docs/developer/thebuild.html). Since I don't need node (and usually don't want to have it on my machine) I tried to use a docker image like:
`docker run -it --rm -v $(pwd):/usr/src/app -w /usr/src/app node:12 npm install`
but that did not immediately work as I expected (yet).
- `sbt it:test` either should ensure that `samples/js-shopping-cart` is built as a pre-requisite or it should be documented to do so.
- If possible, use a docker image to run node/npm to liberate the user to use any pre-installed node version. This might be too much, but we use docker images for other things for similar reasons.

Let me know what would be appropriate, so I can provide a PR for it if needed.

**Comment by marcellanz** [comment]
I see the same issue reported here too:
https://github.com/lightbend/cloudstate-samples/issues/251

**Comment by pvlugter** [comment]
Agree that needing to run all these npm installs manually is awkward. If we start breaking out the node-support module, it could be contained to just that project and more easily automated. And with the TCK integration tests running against a docker published version.

**Comment by pvlugter** [comment]
Assume this is fixed by #327
